File: packages/ts-sdk/test/adapters/federation.test.ts
import { describe, expect, it } from 'vitest';
import { FederationAdapter } from '../../src/treedx/index.js';
import { MockTransport } from './mock.js';

describe('FederationAdapter', () => {
  it('constructs federation endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new FederationAdapter({ transport });
    await adapter.plan({});
    await adapter.search({});
    await adapter.query({});
    await adapter.contextBuild({});
    await adapter.graphQuery({});
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/federation/query/plan',
      'POST /api/v1/search',
      'POST /api/v1/query',
      'POST /api/v1/context/build',
      'POST /api/v1/graph/query'
    ]);
  });
});

File: packages/elixir-sdk/test/adapters/federation/federation_test.exs
defmodule TreeDxSdk.FederationAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Federation.plan(client, %{})
    TreeDxSdk.Federation.search(client, %{})
    TreeDxSdk.Federation.query(client, %{})
    TreeDxSdk.Federation.context_build(client, %{})
    TreeDxSdk.Federation.graph_query(client, %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/graph/query")
           )
  end
end
